# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/agnostic/AgnosticDecoderModule.h
# Commit: e3f920da2232
# Full Hash: e3f920da22320921473f385cb060ed7978d67df9
# Author: Dan Glastonbury <dan.glastonbury@gmail.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - P3: Only create PDMs that are supported in the current process. r=kamidphish
#   
#   Depends on D52797
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54876
# ==============================================================================

diff -r 44471f99280c -r e3f920da2232 dom/media/platforms/agnostic/AgnosticDecoderModule.h
--- a/dom/media/platforms/agnostic/AgnosticDecoderModule.h	Tue Oct 20 23:23:58 2020 +0000
+++ b/dom/media/platforms/agnostic/AgnosticDecoderModule.h	Tue Oct 20 23:24:27 2020 +0000
@@ -13,12 +13,13 @@
 
 class AgnosticDecoderModule : public PlatformDecoderModule {
  public:
-  AgnosticDecoderModule() = default;
+  static already_AddRefed<PlatformDecoderModule> Create();
 
   bool SupportsMimeType(const nsACString& aMimeType,
                         DecoderDoctorDiagnostics* aDiagnostics) const override;
 
  protected:
+  AgnosticDecoderModule() = default;
   virtual ~AgnosticDecoderModule() = default;
   // Decode thread.
   already_AddRefed<MediaDataDecoder> CreateVideoDecoder(